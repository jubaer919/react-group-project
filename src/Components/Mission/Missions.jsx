import React from 'react';
import Table from 'react-bootstrap/Table';

export default function Missions() {
  return (
    <div className="px-3 container">
      <Table responsive="md" className="my-5" striped>
        <thead className="p-5">
          <tr className="border">
            <th className="border">Mission</th>
            <th className="border">Description</th>
            <th className="border">Status</th>
            <th className="border"> </th>
          </tr>
        </thead>
      </Table>
    </div>
  );
}
