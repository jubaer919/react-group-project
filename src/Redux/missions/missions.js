import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const BASE_URL = 'https://api.spacexdata.com/v3/';
const MISSIONS_URL = `${BASE_URL}missions`;

const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
  const response = await axios.get(MISSIONS_URL);
  return response.data.map((mission) => ({
    id: mission.mission_id,
    name: mission.mission_name,
    description: mission.description,
    reserved: false,
  }));
});

const missionsSlice = createSlice({
  name: 'missions',
  initialState: [],
  reducers: {
    joinMission(state, action) {
      return state.map((mission) => (
        mission.id === action.payload ? { ...mission, reserved: true } : mission
      ));
    },
    leaveMission(state, action) {
      return state.map((mission) => (
        mission.id === action.payload
          ? { ...mission, reserved: false }
          : mission
      ));
    },
  },
  extraReducers: (builder) => {
    builder.addCase(fetchMissions.fulfilled, (state, action) => action.payload);
  },
});

export const { joinMission, leaveMission } = missionsSlice.actions;
export { fetchMissions };
export default missionsSlice.reducer;
