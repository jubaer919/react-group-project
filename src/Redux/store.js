import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './Rocket/rocketSlice';
import missionReducer from './missions/missions';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
