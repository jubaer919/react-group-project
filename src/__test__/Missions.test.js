import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../Components/Mission/Missions';
import store from '../Redux/store';

import { joinMission, leaveMission } from '../Redux/missions/missions';

describe('Missions tests', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should return an action with type missions/joinMission', () => {
    const id = 1;
    const action = joinMission(id);
    expect(action.type).toBe('missions/joinMission');
  });

  it('should should return an action with type missions/leaveMission', () => {
    const id = 1;
    const action = leaveMission(id);
    expect(action.type).toBe('missions/leaveMission');
  });
});
