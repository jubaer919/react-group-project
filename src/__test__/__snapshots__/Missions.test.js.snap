// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests render test 1`] = `
<div
  className="px-3 container"
>
  <div
    className="table-responsive-md"
  >
    <table
      className="my-5 table table-striped"
    >
      <thead
        className="p-5"
      >
        <tr
          className="border"
        >
          <th
            className="border"
          >
            Mission
          </th>
          <th
            className="border"
          >
            Description
          </th>
          <th
            className="border"
          >
            Status
          </th>
          <th
            className="border"
          >
             
          </th>
        </tr>
      </thead>
      <tbody
        className="p-5"
      />
    </table>
  </div>
</div>
`;
